app/ui/src/app/store/step/step.store.ts

```diff
@@ -278,7 +278,8 @@ $\{in.body.title\} // Evaluates true when body contains title.
         s =>
           s.action &&
           s.action.descriptor.outputDataShape &&
-          s.action.descriptor.outputDataShape.kind !== DataShapeKinds.NONE
+          s.action.descriptor.outputDataShape.kind !== DataShapeKinds.NONE &&
+          s.action.descriptor.outputDataShape.kind !== DataShapeKinds.ANY
```

Comment:
Just one thing want to make sure - is this OK for "Describe Data" scenario? `DataShapeKinds.ANY` is kinda intermediate state, once user define a custom DataShape specification via "Describe Data" page, then that DataShape turns into "shaped" state, which is available for adding datamapper step.

Reply:
If this method is not used to determine showing a warning for Describe Data, then that would be fine.

Reply:
Yeah, if the user selects some other kind than "type spec not required" then this field will be like json schema or instance or etc so this check will pass still.